evalml/automl/automl_algorithm/iterative_algorithm.py

```diff
@@ -119,6 +122,8 @@ def add_result(self, score_to_minimize, pipeline):
     def _transform_parameters(self, pipeline_class, proposed_parameters):
         """Given a pipeline parameters dict, make sure n_jobs and number_features are set."""
         parameters = {}
+        if self._pipeline_params:
```

Comment:
Need the `if ..` here so that they are only passed to the pipeline if they are needed